### ISSUE TITLE
Why calculate logits_per_image  and logits_per_text separately.

### ISSUE THREAD

**Issue by vinson2233** [openai/CLIP#84]
Just a quick question on calculating the logits.

https://github.com/openai/CLIP/blob/main/clip/model.py line 363
```python 
logit_scale = self.logit_scale.exp()  
logits_per_image = logit_scale * image_features @ text_features.t()  
logits_per_text = logit_scale * text_features @ image_features.t()  
```
Let 
BS = Batch_size 
A = image_features is matrix with dimension BS x 512
B = text_features is matrix with dimension BS x 512  
x' is transpose of x
Then AxB' = (BxA')'

Is there any reason to not writing logits_per_text simply as the transpose of logits_per_image?
```python
logits_per_text = logits_per_image.t()
```
Also in the paper you only calculate the logits once
logits = np.dot(I_e, T_e.T) * np.exp(t) 

Thanks

**Comment by jongwook** [comment]
That's correct, no need to do the same calculation twice. The two lines were remnant of distributed training where the transposed matrices were all-gathered tensors.

**Comment by vinson2233** [comment]
I see, thanks :)
